src/pages/Lending/Pool/components/Repay/RepayInput.tsx

```diff
@@ -131,27 +131,32 @@ export const RepayInput = ({
     enabled: !!repaymentAsset?.assetId,
   })
 
-  const { data: allowanceData, isLoading: isAllowanceDataLoading } = useAllowance({
+  const {
+    approveMutation: {
+      mutate,
+      isPending: isApprovalMutationPending,
+      isSuccess: isApprovalMutationSuccess,
+    },
+    approvalFeesQuery: { data: approvalFeesData, isLoading: isApprovalFeesDataLoading },
+    approvalTxId,
+    setApprovalTxId,
+    isApprovalRequired,
+    allowanceDataQuery: { isLoading: isAllowanceDataLoading },
+  } = useApprove({
     assetId: repaymentAsset?.assetId,
     spender: inboundAddressData?.router,
     from: userAddress,
+    amountCryptoBaseUnit:
+      confirmedQuote?.repaymentAmountCryptoPrecision && repaymentAsset
+        ? toBaseUnit(
+            // Add 5% buffer to the repayment allowance to avoid asset rates fluctuations ending up in more asset needed to repay
```

Comment:
Comment/logic just moved around but while at it, noticed we actually ditched it at some point for some reason, which broke things